File: src/15_sea_battle/​​battle.cpp
#include <iostream>

using namespace std;

const int FIELD = 102;

int main(void) {
    int n;
    cin >> n;
    int a[FIELD][FIELD] = {0};
    for (int i = 0; i < n; ++i) {
        int x1, y1, x2, y2;
        cin >> x1 >> y1 >> x2 >> y2;
        for (int x = x1 - 1; x <= x2 + 1; ++x) {
            for (int y = y1 - 1; y <= y2 + 1; ++y) {
                if (a[x][y] == 1) {
                    cout << "BAD" << endl;
                    return 0;
                }
                a[x][y] == 2;
            }
        }
        for (int x = x1; x <= x2; ++x) {
            for (int y = y1; y <= y2; ++y) {
                a[x][y] = 1;
            }
        }
    }
    cout << "OK" << endl;
    return 0;
}

File: src/15_sea_battle/​​battle.py
FIELD = 102

a = [[0] * FIELD] * FIELD
n = int(input())
for _ in range(n):
    x1, y1, x2, y2 = map(
        int, input().split())
    if x1 > x2:
        x1, x2 = x2, x1
    if y1 > y2:
        y1, y2 = y2, y1
    for x in range(x1 - 1, x2 + 2):
        for y in range(y1 -1, y2 + 2):
            if a[x][y] == 1:
                print("BAD")
                exit()
            a[x][y] = 2
    for x in range(x1, x2 + 1):
        for y in range(y1, y2 + 1):
            a[x][y] = 1
print("OK")